### DISCUSSION TITLE
PCに接続していない方のキーボードが反応しない

### DISCUSSION THREAD

**Discussion by san-you** [Yowkees/keyball#350] [Q&A]
# 解決したいこと
タイトル通り。
左右とも正常に動作するようにしたいです。

# 前提
- Type-CコネクタのPro Micro を利用
- フリマアプリで購入した黒い基盤(`keyball61 rev1`と記入あり)を利用
- 右手トラックボール

# 現状
- PCにUSB接続した方しか動作しない
  - 右をつないだらトラックボールと右側のキー入力が、左をつないだら左側のキー入力だけが出来る
    - つないでいる方はREMAPで全キーの入力を確認できました
  - 接続時のOLEDはPCに接続している方に入力キーなどの表示、もう片方に Keyballのロゴの表示がされている状態
    - PCに接続する方を入れ替えてもOLEDの表示は入れ替わっています
    - 接続中は赤いLEDがどちらも付きっぱなしです

# 基盤の状態
汚いですが一応DATA1同士の導通は確認しております。

右(トラックボール側):
![IMG_8642](https://github.com/Yowkees/keyball/assets/50412560/93e53a7f-8b3d-44d9-8bdb-b13552d7d21a)

左:
![IMG_8641](https://github.com/Yowkees/keyball/assets/50412560/1f313837-e8bc-402e-b84c-7b9581ac9f41)

# 試したこと
- 過去のDiscussionsの回答の確認
  - 以下を参考に半田の付けなおし、TRSケーブルの変更、導通チェック等を行いました
  - https://github.com/Yowkees/keyball/discussions/134
  - https://github.com/Yowkees/keyball/discussions/161
  - https://github.com/Yowkees/keyball/discussions/245
  - https://github.com/Yowkees/keyball/discussions/301
  - https://github.com/Yowkees/keyball/discussions/313
- ファームウェアの書き換え
  - 0.18.xだとQMK MYMSでのファームウェアのビルドが失敗したので、0.22.3でファームウェアのビルドを行い書き込みました
    - 自分と全く同じ症状の別のキーボードが`#define SPLIT_USB_DETECT`を書き込むと動くようになる、という[記事](https://www.eisbahn.jp/yoichiro/2021/03/split_usb_detect.html#gsc.tab=0)を見て試しました

# 懸念点
- 電子工作が初めてなので当たり前のことが出来ていないかもしれません
- フリマアプリで買った黒い基盤を利用しています
  - Keyballとちゃんと記載があったので古いバージョンなのかなと思い購入しましたが、rev1の基盤だと物理的？に解決不可能なのであればrev2の基盤を作成しなおそうと思っております。
  - その際は大変申し訳ありませんがダイオードなどの購入を検討させて下さい。
- フラックスリムーバーを利用していないので基盤が汚いです

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

お写真を拝見し、まず気になったのが下記写真の赤丸部分が隣の端子とショートしていないでしょうか？（光の加減でそのように見えているだけでしたらすみません）
この端子がちょうどDATA線ですので気になりました。

すでに左右のDATA端子同士の導通をご確認いただいているようですので、次は隣の端子とショートしていないかご確認いただければ幸いです。

その他、同様の不具合で多いのはTRSコネクタの半田付けでして、下記緑丸の端子になります。
念のため、こちらの半田付けも見直していただき、半田が基板のパッドに馴染んで山型になっているかご確認ください。

以上、お手数ですがご確認お願いいたします。

![KB61_datafix](https://github.com/Yowkees/keyball/assets/84209772/35035168-f094-4a87-85f1-00580542184f)

**Reply by san-you** [reply]
回答ありがとうございます。

Data,VCC, GNDの導通を確認しながら不安定なところをはんだし直したら接続出来るようになりました。

基板が悪いかもとずっと悩んで原因にあたりをうまくつけられなかったので助かりました。

ありがとうございます。

**Comment by Yowkees** [comment]
無事に動くようになったようで良かったです。安心しました。
今後ともどうぞよろしくお願いいたします。
